Update canonical from `v0.5` to `v0.6`

vlopes11 opened the pull request:
Canonical 0.6 removes the requirement of store annotation which
facilitates the development an simplifies the annotation.

Resolves: #14

vlopes11 committed:
Update canonical from `v0.5` to `v0.6`

Canonical 0.6 removes the requirement of store annotation which
facilitates the development an simplifies the annotation.

Resolves #14

CPerezz commented on src/annotation.rs:
```diff
@@ -6,125 +6,95 @@
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_leaf(leaf);
+        let cardinality =
+            <Cardinality as Annotation<Leaf<K, V>>>::from_leaf(leaf);
+        let max = <MaxKey<K> as Annotation<Leaf<K, V>>>::from_leaf(leaf);
```
You should be able to reduce boilerprate here:

```suggestion
        let cardinality =
            Cardinality::from_leaf(leaf);
        let max = MaxKey::from_leaf(leaf);
```

vlopes11 replied:
Fixed, was using the old required syntax

CPerezz commented on src/annotation.rs:
```diff
@@ -6,125 +6,95 @@
+            MapAnnotationDefault<K>,
+        >>::combine(node);
 
-        Self {
-            cardinality,
-            max,
-            store: PhantomData,
-        }
+        let max = <MaxKey<K> as Combine<
+            KelvinMap<K, V, MapAnnotationDefault<K>>,
+            MapAnnotationDefault<K>,
+        >>::combine(node);
+
+        Self { cardinality, max }
```
Same as above. You can reduce the code complexity:
```suggestion
let cardinality = Cardinality::combine(node);
        let max = MaxKey::combine(node);

        Self { cardinality, max }
```

vlopes11 replied:
Fixed, was using the old required syntax

CPerezz commented on src/lib.rs:
```diff
@@ -8,6 +8,7 @@
 #![feature(external_doc)]
 #![doc(include = "../README.md")]
 #![warn(missing_docs)]
+#![feature(ordering_helpers)]
```
This is not needed anymore in the latest nightly. If it's kept just for the formatter is fine, otherways, it can be removed since is stable since `1.53`

vlopes11 replied:
I just updated the nightly and the feature is still required

CPerezz reviewed (approved):
A few nits. 
Other than that, LGTM

moCello reviewed (approved):
Looks good to me!

ZER0 reviewed (approved):
LGTM, but you should add the deprecated message in the README and point to `dusk-hamt` repo as we discussed.

vlopes11 committed:
Deprecate the crate in favor of `dusk-hamt`

dusk-hamt contains the latest optimizations of microkelvin and
superseeds this crate.
